### PR TITLE
Panzer MiniEM: Add Darcy problem

### PR DESCRIPTION
**Pull request by cgcgcg**
@trilinos/panzer 

## Motivation
Adds assembly and solution of a Darcy problem to MiniEM. This makes sense, as lots of setup and solver code can be reused. We can think about renaming MiniEM later.

### CODE REVIEW

File: packages/panzer/mini-em/src/closures/MiniEM_RandomScalarForcing_impl.hpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+void RandomScalarForcing<EvalT,Traits>::evaluateFields(typename Traits::EvalData workset)
+{
+  using panzer::index_t;
+
+  // double time = workset.time;
+  auto current_h = Kokkos::create_mirror_view(current.get_static_view());
+
+  if (ir_dim == 3) {
+    for (index_t cell = 0; cell < workset.num_cells; ++cell) {
+      for (int point = 0; point < current_h.extent_int(1); ++point) {
```
Comment (rppawlo) [inline packages/panzer/mini-em/src/closures/MiniEM_RandomScalarForcing_impl.hpp:61]: These functions are not on device. Should they be moved there for efficiency?

**Review by rppawlo — APPROVED** [review]
@cgcgcg - looks good to me!

File: packages/panzer/mini-em/example/BlockPrec/MiniEM_helpers.cpp [hunk 2]
```diff
@@ -0,0 +1,650 @@
...
+      //     TEUCHOS_ASSERT(false);
+      //   physicsEqSet.set("Inverse Permeability", "1/mu");
+      // }
+
+      closure_models.sublist(modelID).sublist("1/dt").set<double>("Value",1.0/dt);
+      // if (physicsEqSet.isType<std::string>("Current")) {
+      //   std::string paramLabel = physicsEqSet.get<std::string>("Current");
+      //   if (closure_models.sublist(modelID).sublist(paramLabel).get<std::string>("Type") == "GAUSSIAN PULSE")
+      //     closure_models.sublist(modelID).sublist(paramLabel).set<double>("dt",dt); // set pulse width such that dt resolves it
+      // }
```
[the remark this answers is not in the document]
Comment (cgcgcg) [reply packages/panzer/mini-em/example/BlockPrec/MiniEM_helpers.cpp:254]: Yes

File: packages/panzer/mini-em/src/closures/MiniEM_RandomScalarForcing_impl.hpp (hunk 1 above)
Comment (cgcgcg) [reply packages/panzer/mini-em/src/closures/MiniEM_RandomScalarForcing_impl.hpp:61]: And yes!

**Comment by cgcgcg** [comment]
Thanks, @rppawlo, for the quick review!

**Comment by sebrowne** [comment]
@cgcgcg this is the first non-me PR that is using the re-enabled Intel configuration.  Just letting you know.  It's passed several times for me in nightly/"timed" testing, so we re-enabled it this morning.

**Comment by cgcgcg** [comment]
Lucky me :-P
Thanks for the heads-up though, I appreciate it!

**Comment by cgcgcg** [comment]
@sebrowne Intel build looks good!

**Review by rppawlo — APPROVED** [review]

**Pull request merged**